inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/TextFileReader.java

```diff
@@ -140,7 +161,26 @@ public void setWaitMillisecond(long millis) {
 
     @Override
     public String getSnapshot() {
-        return StringUtils.EMPTY;
+        if (Objects.nonNull(localSnapshot)) {
+            return file.getPath() + JOB_OFFSET_DELIMITER_TASK + localSnapshot.getSnapshot();
```

Comment:
It's better to add ut to verify offsets of multi tasks while custom end-of-line symbols

Reply:
another PR cover the UT